addons/purchase_stock/models/stock.py

```diff
@@ -75,9 +76,28 @@ class ReturnPicking(models.TransientModel):
 
     def _prepare_move_default_values(self, return_line, new_picking):
         vals = super(ReturnPicking, self)._prepare_move_default_values(return_line, new_picking)
-        vals['purchase_line_id'] = return_line.move_id.purchase_line_id.id
+        if self.location_id.usage == "supplier":
+            vals['purchase_line_id'], vals['partner_id'] = self._push_prepare_partner_from_chain(return_line.move_id)
         return vals
 
+    def _push_prepare_partner_from_chain(self, move):
```

Comment:
This function could be moved to the `stock.move` model and called on the move directly instead of being passed, you can then reuse it in `stock.rule`